de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java

```diff
@@ -208,18 +221,32 @@ protected abstract void transfer(IProgressMonitor monitor,
         throws IOException, SarosCancellationException;
 
     /**
-     * Cleanup acquired resources during {@link #setup} and {@link #transfer}.
-     *
+     * Cleanup ends the negotiation process, by disabling the project based
+     * queue and removes acquired handlers during {@link #setup} and
+     * {@link #transfer}.
+     * 
      * @param monitor
-     *      mapping from remote project ids to the target local projects
-     *
+     *            mapping from remote project ids to the target local projects
+     * 
      * @param projectMapping
-     *      mapping of projects
+     *            mapping of projects
      */
     protected void cleanup(IProgressMonitor monitor,
         Map<String, IProject> projectMapping) {
         fileReplacementInProgressObservable.replacementDone();
 
+        /*
+         * TODO Queuing responsibility should be moved to Project
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS1135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=164&issues=AWTNsbFFi6fIhhPMrSiA&open=AWTNsbFFi6fIhhPMrSiA)

[ISSUE_KEY]: AWTNsbFFi6fIhhPMrSiA